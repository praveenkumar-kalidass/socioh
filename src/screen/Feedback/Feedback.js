import React from 'react';
import { Card, CardItem, Col, Grid } from 'native-base';

import Praveen from '../../asset/image/praveen.jpeg';
import { CONSTANT } from '../../constant';
import {
  FeedbackContainer,
  Avatar,
  Name,
  SocialIcon,
  IconContainer,
} from './Feedback.style';
import { AUTHOR_SOCIAL_LINKS } from './Feedback.schema';

const Feedback = () => {
  return (
    <>
      <FeedbackContainer>
        <Card>
          <CardItem>
            <Grid>
              <Avatar large source={Praveen} />
              <Col>
                <Name>{CONSTANT.SOCIOH.AUTHOR_NAME}</Name>
                <Grid>
                  <For each="social" index="index" of={AUTHOR_SOCIAL_LINKS}>
                    <IconContainer key={`social_link_${index}`}>
                      <SocialIcon {...social} />
                    </IconContainer>
                  </For>
                </Grid>
              </Col>
            </Grid>
          </CardItem>
        </Card>
      </FeedbackContainer>
    </>
  );
};

export default Feedback;
