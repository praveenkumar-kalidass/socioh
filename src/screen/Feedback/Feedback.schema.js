import { COLOR, ICON } from '../../constant';

export const AUTHOR_SOCIAL_LINKS = [
  {
    color: COLOR.FACEBOOK,
    type: ICON.FONT_AWESOME_FAMILY,
    name: ICON.FACEBOOK,
  },
  {
    color: COLOR.INSTAGRAM,
    type: ICON.FONT_AWESOME_FAMILY,
    name: ICON.INSTAGRAM,
  },
  {
    color: COLOR.TWITTER,
    type: ICON.FONT_AWESOME_FAMILY,
    name: ICON.TWITTER,
  },
];
