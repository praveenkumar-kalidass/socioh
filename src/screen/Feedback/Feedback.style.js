import styled from 'styled-components/native';
import { Text, Thumbnail, Icon, CardItem } from 'native-base';

import { FONT, SPACING } from 'constant';

export const FeedbackContainer = styled.View`
  padding-horizontal: ${SPACING.SPACE_20};
`;

export const CardContent = styled(CardItem)`
  margin: ${SPACING.SPACE_10};
`;

export const Name = styled(Text)`
  font-size: ${FONT.SIZE_24};
  margin-bottom: ${SPACING.SPACE_10};
`;

export const Avatar = styled(Thumbnail)`
  margin-right: ${SPACING.SPACE_10};
`;

export const IconContainer = styled.TouchableOpacity``;

export const SocialIcon = styled(Icon)`
  color: ${({ color }) => color};
`;
