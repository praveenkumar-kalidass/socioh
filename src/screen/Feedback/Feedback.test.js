import React from 'react';
import { render } from '@testing-library/react-native';

import Feedback from './index';

describe('Feedback', () => {
  it('should match snapshot', () => {
    const container = render(<Feedback />);

    expect(container).toMatchSnapshot();
  });
});
