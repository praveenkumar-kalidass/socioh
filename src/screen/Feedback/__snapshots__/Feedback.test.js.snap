// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 20,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Image
          large={true}
          source={
            Object {
              "testUri": "../../../src/asset/image/praveen.jpeg",
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 40,
                "height": 80,
                "width": 80,
              },
              Object {
                "marginRight": 10,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 24,
                  "marginBottom": 10,
                },
              ]
            }
          >
            Praveenkumar Kalidass
          </Text>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#1a77f2",
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 30,
                      },
                      Object {
                        "color": "#000",
                        "fontSize": 28,
                        "width": 35,
                      },
                      Object {
                        "color": "#1a77f2",
                      },
                    ],
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                type="FontAwesome"
              >
                
              </Text>
            </View>
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#c136a6",
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 30,
                      },
                      Object {
                        "color": "#000",
                        "fontSize": 28,
                        "width": 35,
                      },
                      Object {
                        "color": "#c136a6",
                      },
                    ],
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                type="FontAwesome"
              >
                
              </Text>
            </View>
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#209dea",
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 30,
                      },
                      Object {
                        "color": "#000",
                        "fontSize": 28,
                        "width": 35,
                      },
                      Object {
                        "color": "#209dea",
                      },
                    ],
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                type="FontAwesome"
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
